File: app/src/main/java/com/vershininds/mixture/sample/presentation/modules/details/contract/DetailsInteractorContract.java
package com.vershininds.mixture.sample.presentation.modules.details.contract;


import com.vershininds.mixture.interactor.MvpInteractor;

public interface DetailsInteractorContract {

    interface Presenter extends MvpInteractor.Listener {
//        void onObtainedData(@Nullable SampleObject data, @Nullable Throwable throwable);
    }

    interface Interactor extends MvpInteractor<Presenter> {
//        void obtainData();
    }
}

File: app/src/main/java/com/vershininds/mixture/sample/presentation/modules/details/contract/DetailsInteractorContract.kt
package com.vershininds.mixture.sample.presentation.modules.details.contract

import com.vershininds.mixture.interactor.MvpInteractor

interface DetailsInteractorContract {

    interface Presenter : MvpInteractor.Listener

    interface Interactor : MvpInteractor<Presenter>
}